src/test/java/ArrayIndexedBagTest.java

```diff
@@ -1,412 +1,514 @@
+                }
+
+                @ParameterizedTest
+                @CsvSource({"10", "30", "50"})
+                void remove_existingElement_returnsTrue(Integer element) {
+                    assertTrue(classUnderTest.remove(element));
+                }
+
+                @ParameterizedTest
+                @CsvSource({"10", "30", "50"})
+                void remove_existingElement_removesElement(Integer element) {
+                    classUnderTest.remove(element);
+                    assertFalse(classUnderTest.contains(element));
+                }
+
+                @Test
+                void remove_nonexistentElement_throwsNoSuchElementException() {
+                    assertThrows(NoSuchElementException.class, () -> classUnderTest.remove(Integer.valueOf(0)));
+                }
+
+                @ParameterizedTest
+                @CsvSource({"0, 10", "2, 30", "4, 50"})
+                void indexOf_existingElement_returnsCorrectIndex(int index, Integer element) {
+                    assertEquals(index, classUnderTest.indexOf(element));
+                }
+
+                @Test
+                void indexOf_nonexistentElement_throwsNoSuchElementException() {
+                    assertThrows(NoSuchElementException.class, () -> classUnderTest.indexOf(0));
+                }
+
+                @ParameterizedTest
+                @CsvSource({"10", "30", "50"})
+                void contains_existingElement_returnsTrue(Integer element) {
+                    assertTrue(classUnderTest.contains(element));
+                }
+
+                @Test
+                void contains_nonexistentElement_returnsFalse() {
+                    assertFalse(classUnderTest.contains(0));
+                }
+
+                @ParameterizedTest
+                @CsvSource({"10", "30", "50"})
+                void count_existingElement_returnsOne(Integer element) {
+                    assertEquals(1, classUnderTest.count(element));
+                }
+
+                @Test
+                void count_nonexistentElement_returnsZero() {
+                    assertEquals(0, classUnderTest.count(0));
+                }
+
+                @Test
+                void isEmpty_many_returnsFalse() {
+                    assertFalse(classUnderTest.isEmpty());
+                }
+
+                @Test
+                void size_many_returnsCorrectSize() {
+                    assertEquals(5, classUnderTest.size());
+                }
+
+                @Test
+                void iterator_many_returnsElementsInCorrectOrder() {
+                    Iterator<Integer> iterator = classUnderTest.iterator();
+                    List<Integer> expected = List.of(10, 20, 30, 40, 50);
+                    List<Integer> list = new ArrayList<>();
+                    while (iterator.hasNext()){
+                        list.add(iterator.next());
+                    }
+                    assertEquals(expected, list);
+                }
```

Comment:
Is this what you had in mind when saying:

> ... or just collect the iterator into a list and assert equality on the list.

The singleton and many do not check for the exception, but the empty collection iterator test checks that.

Reply:
That is what I meant, yeah. You can clean it up a tiny bit if you prefer:
```java
List<T> list = new ArrayList<>();
classUnderTest.iterator().forEachRemaining(list::add);
assertEquals(list, List.of(1, 2, 3, 4));
```